validation/src/main/java/io/micronaut/validation/exceptions/ConstraintExceptionHandler.java

```diff
@@ -46,14 +47,20 @@
 @Requires(classes = {ConstraintViolationException.class, ExceptionHandler.class})
 public class ConstraintExceptionHandler implements ExceptionHandler<ConstraintViolationException, HttpResponse<JsonError>> {
 
+    private final JacksonConfiguration jacksonConfiguration;
+
+    public ConstraintExceptionHandler(JacksonConfiguration jacksonConfiguration) {
+        this.jacksonConfiguration = jacksonConfiguration;
+    }
+
     @Override
     public HttpResponse<JsonError> handle(HttpRequest request, ConstraintViolationException exception) {
         Set<ConstraintViolation<?>> constraintViolations = exception.getConstraintViolations();
         if (constraintViolations == null || constraintViolations.isEmpty()) {
             JsonError error = new JsonError(exception.getMessage() == null ? HttpStatus.BAD_REQUEST.getReason() : exception.getMessage());
             error.link(Link.SELF, Link.of(request.getUri()));
             return HttpResponse.badRequest(error);
-        } else if (constraintViolations.size() == 1) {
+        } else if (constraintViolations.size() == 1 && !jacksonConfiguration.getHateoas().isAlwaysSerializeErrorsAsList()) {
```

Comment:
Please assign `jacksonConfiguration.getHateoas().isAlwaysSerializeErrorsAsList()` to a final local field to avoid repeated method calls